joomla/joomla-cms
# [5.4] [#43505] Fixed not showing the changelog URL contents

Pull Request for Issue #43505.

### Summary of Changes
The transport options write the body to the stream interface, by doing so the pointer gets placed at the end of the stream. Any code that uses the `Response` class to get the body will get an empty string instead of the read data. So we need to rewind the cursor before returning the response.

In addition the Changelog class should not access the body directly but use the `getBody()->getContents()` function.


### Testing Instructions
1. Setup the changelog feature as explained in https://manual.joomla.org/docs/building-extensions/install-update/installation/change-log/ for any extension in your site
2. Click on the version number in the extensions manage list
3. Notice that only the modal opens with the title
4. Apply the patch
5. Click on the version number again
6. Notice that only the modal opens with the title and the changelog

#### Alternative testing
##### Manage extension
1. Unzip the attached XML file and place it in a location the test site can reach it via a URL. I assume you place it in the root of the website.
[changelog.zip](https://github.com/user-attachments/files/18195228/changelog.zip)
2. For the database run this query, and replace `jos_` with your prefix and `DOMAIN` with your domain
`UPDATE jos_extensions SET changelogurl = 'https://DOMAIN/changelog.xml' WHERE `element` = 'com_actionlogs' and `type` = 'component';`
3. Go to `System` -> `Manage` -> `Extensions`
4. Filter the list on `action` and type `component`
5. You should see the following:
![image](https://github.com/user-attachments/assets/7b575a7c-9daf-4126-8003-0d48b9399b16)
6. Click on the version number `3.9.0` that will show the modal with the changelog
![image](https://github.com/user-attachments/assets/05f8bda4-8677-4ed9-98bb-50c78c4a9bff)

##### Update extension
1. Unzip the attached XML file and place it in a location the test site can reach it via a URL. I assume you place it in the root of the website.
[changelog.zip](https://github.com/user-attachments/files/18195228/changelog.zip)
2. This is a bit tricky because you need to have an extension with an update available. Therefore make sure you have an outdated extension in your site.
3. Go to `System` -> `Update` -> `Extensions`
4. If needed click on `Check for Updates`
5. Check the `jos_updates` table (replace `jos_` with your own prefix) for the entry that is about the outdated extension
6. Replace `version` field with `3.9.0`
7. Replace the `changelogurl` field with the URL where you placed the changelog XML file
8. Replace the `element` field with `com_actionlogs`
9. Refresh the list of outdated extensions
10. You will now see the `Changelog` button next to the extension you modified and the available version will show `3.9.0` because that is what we changed
![image](https://github.com/user-attachments/assets/4fb86648-e14b-4ac1-a325-cf84023ba5cc)
11. Click on the changelog button to see the modal with changes
![image](https://github.com/user-attachments/assets/75742002-4ffd-47e8-92e7-372a94602a98)


Before:
![image](https://github.com/user-attachments/assets/06acdb7b-1465-44d3-8cea-748f4446541f)

After:
![image](https://github.com/user-attachments/assets/0fb80e6d-b6bb-4bd2-83f6-7f0fea9dcb41)


### Actual result BEFORE applying this Pull Request
A modal is shown with only the title


### Expected result AFTER applying this Pull Request
The modal is shown with the changelog entries


### Link to documentations
Please select:
